src-tools/netutils/test_speed_packet_size.py

```diff
@@ -125,30 +123,40 @@ def parse_args():
                                                      'of packet\n Before run script you have to build '
                                                      'program first')
     arg_parser.add_argument('--cmd', action='append', type=str, required=True,
-			help='my_p2p_addr:port,rpc:rpc_port ; example "10.0.0.1:7000,192.168.1.5:9000" and then --cmd "sameboth,sameip:9001" '
-			'This means: spam me on IP 10.0.0.1:7000, and this RPC request is sent to 192.168.1.5 on port 9000 and 9001 ; '
-		  'This option can be given several times on cmd line, each time you give "endpoint1,endpoint2" ; '
-			'The endpoint is in form of "ip:port" ; Instead of given endpoint you can write "sameboth" or "sameip:sameport" then IP, PORT or both are copied from preceding --cmd option ; '
-			)
+            help='my_p2p_addr:port,rpc:rpc_port ; example "10.0.0.1:7000,192.168.1.5:9000" and then --cmd "sameboth,sameip:9001" '
+            'This means: spam me on IP 10.0.0.1:7000, and this RPC request is sent to 192.168.1.5 on port 9000 and 9001 ; '
+          'This option can be given several times on cmd line, each time you give "endpoint1,endpoint2" ; '
+            'The endpoint is in form of "ip:port" ; Instead of given endpoint you can write "sameboth" or "sameip:sameport" then IP, PORT or both are copied from preceding --cmd option ; '
+            )
+    arg_parser.add_argument('--ranges', type=int, required=True, help='How detailed range should we take: 0=very few, 10=normal, 20=more, 100=everything(slow)')
     arg_parser.add_argument('--passwd', type=str, required=True, help='The RPC password that you set on the remote sender programs')
 
     return arg_parser.parse_args()
 
+# range low..high, but only integers divisible by 'mod' value.
 def myrange(low,high,mod):
-  low2 = (low//mod)*mod
-  high2 = (high//mod)*mod
-  return range(low2, high2, mod)
+    low2 = (low//mod)*mod
+    high2 = (high//mod)*mod
+    return range(low2, high2, mod)
+
 
 if __name__ == "__main__":
+    print("\nPython Script: Starting the wire/packet-size tests.")
+    print("Python Script: run me with option --h to see help.\n")
     args = parse_args()
     PASSWORD = args.passwd
-    # sizes = sorted(  set(list(range(64, 2000, 16)) + list(range(2000, 9500, 64)))  )
-    # sizes = sorted(set(list(range(64, 9500, 64))))
-    sizes = sorted(set(
-      list(myrange(0, 1500, 64))
-      + list(myrange(1500,9000,128))
-      + list([1472,8972])
-    ))
+    ranges = args.ranges
+    sizes = list([256,1472,8972])
+
+    if ranges >= 5 :
+        rtab=(512, 2048)
+        if (args.ranges >= 10): rtab=(128,512)
+        if (args.ranges >= 20): rtab=(32,64)
+        if (args.ranges >= 25): rtab=(8,16)
+        if (args.ranges >= 30): rtab=(4,4)
+        if (args.ranges >= 100): rtab=(1,1)
```

Comment:
I think it would be better:
if (args.ranges >= 100): rtab=(1,1)
elif (args.ranges >= 30): rtab=(4,4)
elif (args.ranges >= 25): rtab=(8,16)
elif (args.ranges >= 20): rtab=(32,64)
elif (args.ranges >= 10): rtab=(128,512)